### PR TITLE
fix(carbon-react): removes default props for accessible placeholders

### LINKED ISSUE #9466: Do not provide accessible labels by default, for example: "Provide icon description"
There are several situations where we use `defaultProps` to provide a placeholder accessible label. In v11, these props should be treated as required with no default props.

```jsx
function TestComponent() {
  // ...
}

// From
TestComponent.propTypes = {
  iconDescription: PropTypes.string,
};

TestComponent.defaultProps = {
  iconDescription: 'Provide a description',
};

// To
TestComponent.propTypes = {
  iconDescription: PropTypes.string.isRequired,
};

TestComponent.defaultProps = {
};
```

This may be tricky to get right with feature flags and most likely will require using the top-level feature flag package instead of being able to use a hook 😞 

Below is a list of components that currently do this:

- [ ] FileUploader
- [ ] SelectItemGroup
- [ ] NumberInput
- [ ] Switch
- [ ] OverflowMenuItem

Note: there may have been some that have been missed, this was a quick search!

### PR DESCRIPTION
**Pull request by abbeyhrt**
Closes #9466
(Once #9731 is merged) 

Removes default props in v11 for accessible placeholder's and ensures that the props are required. 

Also makes `Tab`'s `id` prop required since the component has an accessibilty violation if it's not provided (reference #9666) 

#### Changelog

**Changed**

- iconDescription for `FileUploader` is now required and default is empty
- iconDescription's and invalidText's default prop for `NumberInput` are now empty
- default prop for label is now empty and `label` and `id` are required

#### Testing / Reviewing

Verify that everything works as expected

### CODE REVIEW

File: packages/react/src/components/FileUploader/FileUploader.js [hunk 1]
```diff
@@ -97,7 +100,9 @@ export default class FileUploader extends React.Component {
   static contextType = PrefixContext;
 
   static defaultProps = {
-    iconDescription: 'Provide icon description',
+    iconDescription: FeatureFlags.enabled('enable-v11-release')
+      ? ''
```
Comment (joshblack) [inline packages/react/src/components/FileUploader/FileUploader.js]:
```suggestion
      ? undefined
```

File: packages/react/src/components/NumberInput/NumberInput.js [hunk 2]
```diff
@@ -190,10 +191,14 @@ class NumberInput extends Component {
   static defaultProps = {
     disabled: false,
     hideLabel: false,
-    iconDescription: 'choose a number',
+    iconDescription: FeatureFlags.enabled('enable-v11-release')
+      ? ''
```
Comment (joshblack) [inline packages/react/src/components/NumberInput/NumberInput.js]:
```suggestion
      ? undefined
```

**Review by joshblack — CHANGES_REQUESTED** [review]
I think for the default props we'll want the value to be `undefined` versus an empty string.

For dropping isRequired, is it the case that all of these are no longer required? It seemed like we would still want them but want the user to provide them versus the default props being used. Let me know if I'm misunderstanding 👀

**Comment by abbeyhrt** [comment]
> For dropping isRequired, is it the case that all of these are no longer required? It seemed like we would still want them but want the user to provide them versus the default props being used. Let me know if I'm misunderstanding 👀

I think you might have it backwards! If v11 is enabled, they are required, if not, they aren't

**Review by aledavila — APPROVED** [review]

File: packages/react/src/components/Tab/Tab.js [hunk 3]
```diff
@@ -98,7 +103,7 @@ export default class Tab extends React.Component {
 
   static defaultProps = {
     role: 'presentation',
-    label: 'provide a label',
+    label: FeatureFlags.enabled('enable-v11-release') ? '' : 'provide a label',
```
Comment (joshblack) [inline packages/react/src/components/Tab/Tab.js]:
```suggestion
    label: FeatureFlags.enabled('enable-v11-release') ? undefined : 'provide a label',
```

File: packages/react/src/components/Tab/Tab.js [hunk 4]
```diff
@@ -44,7 +45,9 @@ export default class Tab extends React.Component {
     /**
      * The element ID for the top-level element.
      */
-    id: PropTypes.string,
+    id: FeatureFlags.enabled('enable-v11-release')
+      ? PropTypes.string.isRequired
+      : PropTypes.string,
```
Comment (joshblack) [inline packages/react/src/components/Tab/Tab.js]: Do we auto-generate the id for the component if it doesn't exist already?

File: packages/react/src/components/SelectItemGroup/SelectItemGroup.js [hunk 5]
```diff
@@ -55,7 +56,7 @@ SelectItemGroup.propTypes = {
 
 SelectItemGroup.defaultProps = {
   disabled: false,
-  label: 'Provide label',
+  label: FeatureFlags.enabled('enable-v11-release') ? '' : 'Provide label',
```
Comment (joshblack) [inline packages/react/src/components/SelectItemGroup/SelectItemGroup.js]:
```suggestion
  label: FeatureFlags.enabled('enable-v11-release') ? undefined : 'Provide label',
```

File: packages/react/src/components/NumberInput/NumberInput.js [hunk 6]
```diff
@@ -190,10 +191,14 @@ class NumberInput extends Component {
   static defaultProps = {
     disabled: false,
     hideLabel: false,
-    iconDescription: 'choose a number',
+    iconDescription: FeatureFlags.enabled('enable-v11-release')
+      ? undefined
+      : 'choose a number',
```
Comment (joshblack) [inline packages/react/src/components/NumberInput/NumberInput.js:196]: Should we just update `iconDescription` to `label` in a general sense for components that require labels?

File: packages/react/src/components/NumberInput/NumberInput.js [hunk 7]
```diff
@@ -190,10 +191,14 @@ class NumberInput extends Component {
...
     step: 1,
     invalid: false,
-    invalidText: 'Provide invalidText',
+    invalidText: FeatureFlags.enabled('enable-v11-release')
+      ? ''
```
Comment (joshblack) [inline packages/react/src/components/NumberInput/NumberInput.js]:
```suggestion
      ? undefined
```

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/react/src/components/NumberInput/NumberInput.js (hunk 6 above)
Comment (abbeyhrt) [reply packages/react/src/components/NumberInput/NumberInput.js:196]: I think a lot of the time, the iconDescription is a distinct thing from the label and depends component to component, so it's something we could talk about but I'm not sure if it should prevent this PR from being merged in

**Review by joshblack — APPROVED** [review]

**Pull request merged**